### PR TITLE
MM-14768: Switches field type from sql.NullBool to a bool pointer.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary

Switches field type from sql.NullBool to a bool pointer.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-14768

### CODE REVIEW

File: model/channel_test.go [hunk 1]
```diff
@@ -59,8 +59,8 @@ func TestChannelPatch(t *testing.T) {
 	if *p.Purpose != o.Purpose {
 		t.Fatal("do not match")
 	}
-	if *p.GroupConstrained != o.GroupConstrained.Bool {
-		t.Fatalf("expected %v got %v", *p.GroupConstrained, o.GroupConstrained.Bool)
+	if *p.GroupConstrained != o.GroupConstrained {
```
Comment (jespino) [inline model/channel_test.go]: I guess you need here the `*o.GroupConstrained` like in the other test

Comment (mkraft) [inline model/channel_test.go]:
```suggestion
	if *p.GroupConstrained != *o.GroupConstrained {
```

File: model/team_test.go [hunk 2]
```diff
@@ -171,7 +171,7 @@ func TestTeamPatch(t *testing.T) {
 	if *p.AllowOpenInvite != o.AllowOpenInvite {
 		t.Fatal("AllowOpenInvite did not update")
 	}
-	if *p.GroupConstrained != o.GroupConstrained.Bool {
-		t.Fatalf("expected %v got %v", *p.GroupConstrained, o.GroupConstrained.Bool)
+	if *p.GroupConstrained != *o.GroupConstrained {
+		t.Fatalf("expected %v got %v", *p.GroupConstrained, o.GroupConstrained)
```
Comment (mkraft) [inline model/team_test.go]:
```suggestion
		t.Fatalf("expected %v got %v", *p.GroupConstrained, *o.GroupConstrained)
```

File: model/channel_test.go [hunk 3]
```diff
@@ -59,8 +59,8 @@ func TestChannelPatch(t *testing.T) {
 	if *p.Purpose != o.Purpose {
 		t.Fatal("do not match")
 	}
-	if *p.GroupConstrained != o.GroupConstrained.Bool {
-		t.Fatalf("expected %v got %v", *p.GroupConstrained, o.GroupConstrained.Bool)
+	if *p.GroupConstrained != *o.GroupConstrained {
+		t.Fatalf("expected %v got %v", *p.GroupConstrained, o.GroupConstrained)
```
Comment (mkraft) [inline model/channel_test.go]:
```suggestion
		t.Fatalf("expected %v got %v", *p.GroupConstrained, *o.GroupConstrained)
```

**Comment by jespino** [comment]
I'm thinking now, if we use `bool` instead of `*bool`the `NULL` value from the database wouldn't be deserialized to `false` (the default nil value for bool)?

**Comment by mkraft** [comment]
@jespino 

```
sql: Scan error on column index 15, name "groupconstrained": sql/driver: couldn't convert <nil> (<nil>) into type bool
```

**Pull request merged**